File: design-patterns/ts/visitor/hosting-cost/compute-service.ts
// compute-service.ts

import HostingCalculatorVisitor from "./hosting-calculator-visitor";
import Service from "./service";


class ComputeService implements Service {

    private readonly price: number = 10.50;
    private quantity: number;

    constructor(quantity: number) {
        this.quantity = quantity;
    }

    getPrice(): number {
        return this.price;
    }

    getQuantity(): number {
        return this.quantity;
    }

    accept(hostingCalculatorVisitor: HostingCalculatorVisitor): number {
        return hostingCalculatorVisitor.visit(this);
    }
}

export default ComputeService;

File: design-patterns/go/visitor/hosting_cost/compute_service.go
// compute_service.go

package main

type ComputeService struct {
	price    float64
	quantity int
}

func NewComputeService(quantity int) (compute *ComputeService) {
	compute = &ComputeService{}
	compute.price = 10.50
	compute.quantity = quantity
	return
}

func (compute *ComputeService) GetPrice() float64 {
	return compute.price
}

func (compute *ComputeService) GetQuantity() int {
	return compute.quantity
}

func (compute *ComputeService) Accept(hostingCalculatorVisitor HostingCalculatorVisitor) float64 {
	return hostingCalculatorVisitor.ComputeVisit(compute)
}
